inductiva/resources/machines.py

```diff
@@ -0,0 +1,112 @@
+"""Containes MachineGroup class to manage GC resources."""
+from absl import logging
+from typing import Optional
+import uuid
+import inductiva
+from inductiva import api
+from inductiva import resources
+
+from inductiva.client import ApiClient, ApiException
+from inductiva.client.apis.tags.instance_api import InstanceApi
+from inductiva.client.model.instance import Instance
+from inductiva.client.model.instance_group import InstanceGroup
+
+
+class MachineGroup():
+    """Class to manage Google Cloud resources."""
+
+    def __init__(
+        self,
+        machine_type: str,
+        num_machines: int = 1,
+        spot: bool = False,
+        disk_size_gb: int = 30,
+        label: str = None,
+        machine_group_id: Optional[uuid.UUID] = None,
+    ) -> None:
+        """Create a MachineGroup object.
+
+        Args:
+            machine_type: The type of GC machine to launch. Ex: "e2-medium".
+            num_machines: The number of machines to launch.
+            spot: Whether to use spot instances.
+            disk_size_gb: The size of the disk in GB. (min. 30 GB)
+            label: The label to assign to the machine group.
+            machine_group_id: If None the Machine Group ID will be generated
+              automatically. A new ID can be created by invoking the
+              inductiva.resources.create_resource_pool() method.
```

Comment:
What is the goal of being able to set the machine group ID when creating a new machine group? This kind of ID is usually automatically generated (to guarantee uniqueness, etc.).

Reply:
The goal was to create and ID for the machines and send all the tasks to that specific machine group. Ex: when a user wants to create two consistent machine groups and send the tasks to a specific group, they can use this ID explicitly.

Reply:
Yes, I would argue for the same workflow, but this is how we started it initially. I will leave it as it is for now and will bring it up to discussion in the meeting.

Reply:
I think it's better if you simply use it internally and don't expose it in the MachineGroup API. Later if we implement it differently there will be no changes to user's interface